### PR TITLE
Updated handle_guided_request() to report error while setting guided points. 

### PR DESCRIPTION
**Pull request by nitirohilla**
This PR solves the issue report in #3921.

handle_guided_request() is updated to report the error "MAV_MISSION_ERROR" while setting guided points from modes other than guided.

### CODE REVIEW

**Comment by WickedShell** [comment]
Nice, looks good to me. (Have not tested in the simulator but looks exactly like what I was planning to do.)

File: libraries/GCS_MAVLink/GCS_Common.cpp [hunk 1]
```diff
@@ -641,10 +641,13 @@ bool GCS_MAVLINK::handle_mission_item(mavlink_message_t *msg, AP_Mission &missio
     if (packet.current == 2) {                                               
         // current = 2 is a flag to tell us this is a "guided mode"
         // waypoint and not for the mission
-        handle_guided_request(cmd);
+        result = MAV_MISSION_ACCEPTED;
+
+        if (!handle_guided_request(cmd)) {
+            result = MAV_MISSION_ERROR;
+        }
```
Comment (khancyr) [inline libraries/GCS_MAVLink/GCS_Common.cpp]: Could be simplify with 
result = (handle_guided_request(cmd) ? MAV_MISSION_ACCEPTED : MAV_MISSION_ERROR) ;
and we gain 1 assigment \o/

**Comment by rmackay9** [comment]
yes, i agree too.  looks like the right thing to do.

**Comment by WickedShell** [comment]
I had the thought, but we do enough of the other style I wasn't going to
harp on it (and I suspect the compiler gets it right). I admit I'd rather
see an if/else then the ternary operator, or the current approach. (My
experience is that it is easy to mess up the ternary operator argument
order or get a paren wrong). Just my two cents.

**Comment by nitirohilla** [comment]
I have done the changes for ternary operator and tested in SITL, it is working fine. I have also updated the pull request for the same.

**Comment by nitirohilla** [comment]
I have done the changes for ternary operator and tested in SITL, it is working fine. I have also updated the PR for the same.

**Comment by khancyr** [comment]
I don't like ternary operator in general, but in this case, I think it fit well and has a better readability.

**Comment by lucasdemarchi** [comment]
Applied, thanks.

**Comment by nitirohilla** [comment]
Thanks @lucasdemarchi.

**Pull request closed without merging**